src/Photos/PHCompat.cs

```diff
@@ -20,6 +21,17 @@ public virtual void SetProgressHandler (PHProgressHandler progressHandler)
 			ProgressHandler = progressHandler;
 		}
 	}
-}
+#endif
 
+#if !XAMCORE_4_0 && !MONOMAC
+	// incorrect signature, should have been `ref NSError`
+	[Obsolete ("Use 'PHLivePhotoFrameProcessingBlock2' instead.")]
+	public delegate CIImage PHLivePhotoFrameProcessingBlock (IPHLivePhotoFrame frame, NSError error);
+
+	public partial class PHLivePhotoEditingContext {
+
+		[Obsolete ("Use 'FrameProcessor2' instead.")]
+		public virtual PHLivePhotoFrameProcessingBlock FrameProcessor { get; set; }
```

Comment:
I wonder if it's possible to make existing code kind of work, by doing something like this:

```csharp
public virtual PHLivePhotoFrameProcessingBlock FrameProcessor {
    get {
         var fp = FrameProcessor2;
         return new PHLivePhotoFrameProcessingBlock ((IPHLivePhotoFrame frame, NSError error) =>
         {
             return fp (frame, ref error);
         });
    }
    set {
        FrameProcessor2 = new PHLivePhotoFrameProcessingBlock2 ((IPHLivePhotoFrame frame, ref NSError error) =>
         {
             return value (frame, error);
         });
    }
}
```

it might not be worth the hassle though.

Reply:
The sample always crash (error or not) with the current code, so throwing should not break anyone. However I can set the obsolete error to true, that will force people to fix it when recompiling (harder to ignore).